### PR TITLE
Integrate missed changes.

### PR DESCRIPTION
**Pull request by chriseth**
It seems that I based the new assembly.rst file off of a really old version of the part inside control structures. This PR moves the changes over and especially also deletes the section inside control structures.

### CODE REVIEW

File: docs/assembly.rst [hunk 1]
```diff
@@ -230,19 +235,22 @@ In the grammar, opcodes are represented as pre-defined identifiers.
 | create(v, p, s)         |      | create new contract with code mem[p..(p+s)) and send v wei      |
 |                         |      | and return the new address                                      |
 +-------------------------+------+-----------------------------------------------------------------+
-| call(g, a, v, in,       |      | call contract at address a with input mem[in..(in+insize)]      |
+| call(g, a, v, in,       |      | call contract at address a with input mem[in..(in+insize))      |
```
Comment (VoR0220) [inline docs/assembly.rst:238]: Should be either a bracket at the end or needs a `(` at the beginning of `mem`

File: docs/assembly.rst [hunk 2]
```diff
@@ -230,19 +235,22 @@ In the grammar, opcodes are represented as pre-defined identifiers.
...
 | insize, out, outsize)   |      | providing g gas and v wei and output area                       |
-|                         |      | mem[out..(out+outsize)] returting 1 on error (out of gas)       |
+|                         |      | mem[out..(out+outsize)) returning 0 on error (eg. out of gas)   |
```
Comment (VoR0220) [inline docs/assembly.rst:240]: ibid

**Review by VoR0220 — CHANGES_REQUESTED** [review]
just change those bits and then this is GTM

File: docs/assembly.rst (hunk 1 above)
Comment (chriseth) [reply docs/assembly.rst:238]: This is an interval that is open at one side, it is explained at the beginning of the table.

Comment (VoR0220) [reply docs/assembly.rst:238]: Oops.

**Review by VoR0220 — APPROVED** [review]

**Pull request merged**